src/components/micro-frame-slot/component/web.component.ts

```diff
@@ -0,0 +1,92 @@
+import StreamGenerator from "../../stream-source/component/StreamGenerator";
+import getWritableDOM from "writable-dom";
+
+interface Input {
+  slotId: string;
+  from: string;
+  catch?: unknown;
+  timeout?: number;
+  loading?: unknown;
+  cache?: RequestCache;
+  headers?: Record<string, string>;
+}
+
+interface State {
+  loading: boolean;
+  err: undefined | Error;
+}
+
+export = {
+  onCreate() {
+    const ssrEl = document.getElementById(this.id);
+    let loading = true;
+    if (ssrEl) {
+      this.slotId = ssrEl.dataset.slotId;
+      this.from = ssrEl.dataset.from;
+      ssrEl.removeAttribute("data-slot-id");
+      ssrEl.removeAttribute("data-from");
+      ssrEl.removeAttribute("id");
+      loading = false;
+    }
+
+    this.state = {
+      loading,
+      err: undefined,
+    };
+  },
+  onMount() {
+    // Only trigger a new load if this wasn't ssr'd, or the src has changed.
+    this.onUpdate();
+  },
+  onDestroy() {
+    this.slot?.done();
+  },
+  async onUpdate() {
+    if (
+      (this.slotId === this.input.slotId && this.from === this.input.from) ||
+      !this.slot
+    )
+      return;
+
+    this.state.loading = true;
+    this.state.err = undefined;
+    this.slotId = this.input.slotId;
+    this.from = this.input.from;
+
+    let writable: ReturnType<typeof getWritableDOM> | undefined;
+    let err: Error | undefined;
+
+    try {
+      writable = getWritableDOM(
+        this.el,
+        // references the start of the preserved Marko fragment.
+        this.el.lastChild!.previousSibling
+      );
+
+      const stream = this.slot?.stream;
```

Comment:
Assuming `stream` will definitely be there by this point we should probably change this to:

```js
const stream = this.slot!.stream;
```